rust/rope/src/rope.rs

```diff
@@ -396,8 +396,24 @@ impl<'de> Deserialize<'de> for Rope {
     where
         D: Deserializer<'de>,
     {
-        let s = String::deserialize(deserializer)?;
-        Ok(Rope::from(s))
+        deserializer.deserialize_str(RopeVisitor {})
```

Comment:
This `{}` is also now unnecessary. I'm actually surprised this compiled, but firing up a playground you can kind of see what's going on. With the three different struct types, the uncommented lines are valid constructors 😁 

```rust
struct Normal{}
struct Tuple();
struct Unit;

fn main() {
    // let n = Normal;
    // let n1 = Normal();
    let n2 = Normal{};
    
    let t = Tuple;
    let t1 = Tuple();
    let t2 = Tuple{};
    
    let u = Unit;
    // let u1 = Unit();
    let u2 = Unit{};
}
```

Reply:
I avoided the "constructors" without bracket/parenthesis previously because they look similar to traits, but I guess that doesn't matter now that dyn is a thing.

Reply:
Traits, even marker traits, require brackets, unless I'm misunderstanding?

```rust
trait MySync; // doesn't compile
trait MySend {} // compiles
```

Reply:
You are right, it was more of a familiarity thing for me.
```rust
// constructor, x an instance of Thing
let x = Thing::new();
// constructor, x an instance of Thing
let x = Thing{};
// x is a reference to the type Thing
let x = Thing;
```
Of course there is not actually an "instance" of Thing because it is optimized away, but all of the languages I have worked with would require an explicit constructor even for a type with no value.

Reply:
Gotcha. Yea, I guess since there aren't type-level functions or higher-kinded types it doesn't really make sense to have a reference to a type, at this point in Rust, but maybe at some point down the road?